bluefog/common/operations.cc

```diff
@@ -63,10 +63,10 @@ void BackgroundThreadLoop(BluefogGlobalState& state) {
   // Initialize controller
   state.controller->Initialize();
 
-#if HAVE_NCCL
-  state.nccl_controller->Initialize();
-  BFLOG(INFO, bluefog_global.controller->GetRank()) << "NCCL Initialized";
-#endif
+// #if HAVE_NCCL
+//   state.nccl_controller->Initialize();
+//   BFLOG(INFO, state.controller->GetRank()) << "NCCL Initialized";
+// #endif
```

Comment:
Why commented out?

Reply:
It is your suggestion, we use lazy initialization. Remove these lines and add a comment.